### ISSUE TITLE
Update applications

### ISSUE THREAD

**Issue by jm42** [strukturart/bHacker-store-client#32]
Now is not possible to install an application that is already installed, even if the last version is newer that the installed version. For a future where the store can update applications, the first step would be showing the last application version from `data.json`.

Then to show the installed version. The [`App` object](https://developer.mozilla.org/en-US/docs/Archive/B2G_OS/API/DOMApplication) has the complete manifest with the version.

It also has a `checkForUpdate()` method.

**Comment by jm42** [comment]
Seems like `checkForUpdate()` might be an Firefox OS only feature.

> Firefox OS on the phone polls once per day for changes to an app. To do this, it polls the URL of the mini-manifest, then polls the URL in the package_path field in the mini-manifest. This is done using the checkForUpdate() method on the App object. When the ETag header changes, it knows the app has been updated. Then it checks if the zip file has changed.

**Comment by strukturart** [comment]
Unfortunately I don't know how to update an app without losing the user data. uninstall-> download new version -> install

**Comment by sunsetsonwheels** [comment]
I think for this to happen, the backend needs to be revamped completely, moving away from the OmniSD format to having mini manifests and application zips: https://developer.mozilla.org/en-US/docs/Archive/Marketplace/API/DOMApplicationsRegistry/install

**Comment by strukturart** [comment]
can we use it to install a certified app?

**Comment by strukturart** [comment]
https://developer.mozilla.org/en-US/docs/Archive/Mozilla/Marketplace/Options/Packaged_apps

**Comment by sunsetsonwheels** [comment]
> can we use it to install a certified app?

I think it should be able to, you could try. I guess this is the same mechanism that the KaiStore uses to install apps behind the scenes.

**Comment by farooqkz** [comment]
> I think for this to happen, the backend needs to be revamped completely, moving away from the OmniSD format to having mini manifests and application zips: https://developer.mozilla.org/en-US/docs/Archive/Marketplace/API/DOMApplicationsRegistry/install

I don't think we need to completely move away from the OmniSD format. OmniSD zip already have application.zip and manifest as long as I remember.

**Comment by strukturart** [comment]
you could try if import () works without OmniSD then you could use both.

**Comment by jm42** [comment]
At lest the store should show last version (and, if installed, the current version.)
